x/auth/ante/sigverify.go

```diff
@@ -204,14 +202,20 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 			return ctx, err
 		}
 
-		signerAccs[i] = acc
-
 		// retrieve pubkey
-		pubKey := signerAccs[i].GetPubKey()
+		pubKey := acc.GetPubKey()
 		if !simulate && pubKey == nil {
 			return ctx, sdkerrors.Wrap(sdkerrors.ErrInvalidPubKey, "pubkey on account is not set")
 		}
 
+		// Check account sequence number.
+		if sig.AccountSequence != acc.GetSequence() {
+			return ctx, sdkerrors.Wrapf(
+				sdkerrors.ErrWrongAccountSequence,
+				"account sequence mismatch, expected %d, got %d", acc.GetSequence(), sig.AccountSequence,
+			)
+		}
```

Comment:
A separate ante handler would also have made sense, but since it's just 7 lines, I thought just to leave it here.

Reply:
Amazing to pull out this error case. Much appreciated

Reply:
Yes, this is the a huge benefit for clients.